### PR TITLE
[PD-1737] Preserve Combobox focus when removing a chip

### PR DESCRIPTION
**Pull request by TheSonOfThomp**
## ✍️ Proposed changes

Ensures the focus remains in the combobox when a chip is removed. Focus is set to either the input, or the next chip to the right, whichever is relevant

🎟 _Jira ticket:_ [PD-1737](https://jira.mongodb.org/browse/PD-1737)

## 🛠 Types of changes

- [ ] Tooling (updates to workspace config or internal tooling)
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Style change (purely visual updates; if this is the case, attach a screenshot below!)

## ✅ Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [x] I have run `yarn changeset` and documented my changes

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README

## Feedback Levels

Please try to mark your comments with one of these levels

1. Praise (👏 🚀 🔥 🍾)
2. Opinions, Ideas, Suggestions (💡 👀)
3. Let’s discuss @ Code review (🤷 🤔 🧐 )
4. Doesn’t meet agreed upon standards (😬 ⚠️ ‼️)
5. Missing an edge case (🛑 🙅 ❌)
6. Missing core functionality (⛔️ ☣️ 🚧 🌋 💀)

(Note: Use ⌘-⌃-␣ to open the Emoji keyboard!)

### CODE REVIEW

File: .changeset/tiny-balloons-count.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@leafygreen-ui/combobox': patch
+---
+
+Ensures the focus remains in the combobox when a chip is removed. Focus is set to either the input, or the next chip to the right, whichever is relevant
```
Comment (shaneeza) [inline .changeset/tiny-balloons-count.md:5]: When you say, `the next chip to the right`, do you mean the last chip? I was testing with five chip and when i removed the first chip the last chip became focused.

Comment (TheSonOfThomp) [reply .changeset/tiny-balloons-count.md:5]: Hmm yeah, I just noticed that too. I'll fix that

Comment (TheSonOfThomp) [reply .changeset/tiny-balloons-count.md:5]: @shaneeza Should be fixed now!

**Review by spark33 — APPROVED** [review]
🔥🔥🔥🔥

**Pull request merged**